### ISSUE TITLE
Contributor Calendar

### ISSUE THREAD

**Issue by parispittman** [apple/swift-org-website#239]
## Is your feature request related to a problem? Please describe

Contributors and various Group/Team maintainers don’t have a Birds Eye view of meetings going on. Some are welcoming others to join and this could be a great way to display contributor community activities. 

Example:
I am a Swift core team member I want to see and interact with a calendar that would have dates of events that appeal to me so that I can be aware and/or attend.

Features: 

* Calendar owners to change their calendar via a YAML file
* Viewers to download an iCal as a subscription so viewers can subscribe to the entire calendar downloading an .ics file and deciding how often to pull for updates

## Describe the solution you'd like

Addition of a calendar to https://www.swift.org/community/ under Community Structure definitions. Open to discussing best location or if a new page would be warranted. 

@federicobucchi would like to lead the development for this and use Stencil and look into various open source packages to deliver the rest. A YAML file would power the information/content. 

## Additional context

Activities to display:

* Workgroup meetings
* Team Meetings
* Release date range 
* Contributor Experience related events like mentoring AMAs and new contributor office hours
* Other future contributor related meetings and events 

What to display

* Info on how to be added to the meetings
* Date/time 
* Related communication channels
* Related meeting notes/forum posts/etc

Other OSS project calendars:

* _https://jupyter.org/community#calendar_ (https://jupyter.org/community#calendar)
* _https://www.kubernetes.dev/resources/calendar/_ (https://www.kubernetes.dev/resources/calendar/)
* _https://lang-team.rust-lang.org/calendar.html_ (https://lang-team.rust-lang.org/calendar.html)
* _https://calendar.google.com/calendar/u/0/embed?src=recud4b9o8cmc0m5rmr033p5nk@group.calendar.google.com_ (https://calendar.google.com/calendar/u/0/embed?src=recud4b9o8cmc0m5rmr033p5nk@group.calendar.google.com)
* _https://community.apache.org/calendars/_ (https://community.apache.org/calendars/)
* _https://github.com/golang/go/wiki/Conferences_ (https://github.com/golang/go/wiki/Conferences) 

## Out of scope
A “community calendar”; this would require a more extensive look at guidelines and process for what is displayed. Definitely think we should go in this direction at some point but this specific use case solves a problem that we can build on top of.

**Comment by 0xTim** [comment]
All for this! One thing to keep in mind (though it shouldn't be a blocker of any sort) is the long term goal for the site is to migrate it to Swift (the exact method TBC). So whilst we don't need to wait for an unknown future it would be good to try and ensure we don't corner ourselves 😅

**Comment by yakatyansh** [comment]
@federicobucchi I would likw to try out this issue you can continue though, I have some experience with react, I just wanted to inform that i am also trying this out

**Comment by yakatyansh** [comment]
@federicobucchi hey i am a student and know some web development, so if possible can i work with you on this issue ?

**Comment by yakatyansh** [comment]
@federicobucchi  thank you so much, looking forward to working with you and learning.

**Comment by yakatyansh** [comment]
@federicobucchi  Until then, I am going through the codebase and taking it all in, if any doubts should i ask you right here ?

**Comment by federicobucchi** [comment]
> @federicobucchi Until then, I am going through the codebase and taking it all in, if any doubts should i ask you right here ?

@yakatyansh I am always learning and checking new things too!  I think that maybe we can leave this Issue to talk specifically about the Calendar and you can use forums for specific frameworks questions!  At some point I will find resources to link!

**Comment by yakatyansh** [comment]
Yeah sure if i find anything useful related to the same i will provide here, to confirm once we are using stencil right ? The web component building framework?

**Comment by yakatyansh** [comment]
Hey @federicobucchi when should we start the work this week ?

**Comment by yakatyansh** [comment]
@federicobucchi i am free now was busy with my final exams untill wednesday, I am now starting with buliding components on stencil and practicing and then i will build a prototype of the calendar and push the same to git

**Comment by yakatyansh** [comment]
Will keep you updated with my progress daily here itself

**Comment by federicobucchi** [comment]
@yakatyansh apologies it's taking some more time.  I am closer to release the first base of web components, but I am figuring some logistics that I will explain later.  Have a nice weekend!

**Comment by yakatyansh** [comment]
@federicobucchi no issues you take your time, have a nice weeknd !!

**Comment by yakatyansh** [comment]
btw @federicobucchi i have sent you a connection request on linkedin i would love to connect with you. My name is Yash Katyan.
